ethcore/src/miner/pool_client.rs

```diff
@@ -120,6 +120,11 @@ impl<'a, C: 'a> PoolClient<'a, C> where
 	pub fn verify_signed(&self, tx: &SignedTransaction) -> Result<(), transaction::Error> {
 		self.engine.machine().verify_transaction(&tx, &self.best_block_header, self.chain)
 	}
+
+	/// Verifies transaction against its block (before its import into this block)
+	pub fn verify_for_pending_block(&self, transaction: &UnverifiedTransaction, header: &Header) -> Result<(), transaction::Error> {
+		self.engine.machine().verify_transaction_basic(transaction, header)
```

Comment:
Why not call `verify_transaction` here as well (and verify against `header` not `best_block_header`)? So that we can actually simplify the miners code and perform only one check.

Reply:
Sorry, I don't understand your suggestion. Which verify_transaction method do you mean? verify_transaction from machine doesn't call required verification methods. verify_transaction from pool client doesn't check against custom header, only for best block's

Reply:
Thanks! Got it.